sklearn/tests/test_base.py

```diff
@@ -600,12 +599,11 @@ def __init__(self, attribute_pickled=5):
         self._attribute_not_pickled = None
 
     def __getstate__(self):
-        data = self.__dict__.copy()
-        data["_attribute_not_pickled"] = None
-        return data
+        state = super().__getstate__()
```

Comment:
Was the original implementation causing the UserWarning?

Reply:
It was causing this warning https://github.com/scikit-learn/scikit-learn/blob/63e158462fdca475215f181bdfc5f732bcb8ae46/sklearn/base.py#L377-L381
That's because it completely overrides the `__getstate__` method so doesn't set the right sklearn version that the `__setsate__` method expects.